src/main/java/core/basesyntax/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+
+        if (currentQuantity == null) {
+            throw new IllegalArgumentException("Fruit does not exist in storage.");
+        }
+        if (currentQuantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity for purchase.");
+        }
+        int purchaseQuantity = transaction.getQuantity();
+        int newQuantity = currentQuantity - purchaseQuantity;
+
+        if (newQuantity < 0) {
+            throw new IllegalArgumentException("Resulting quantity cannot be negative for fruit: "
+                    + transaction);
+        }
+        Storage.storage.put(transaction.getFruit(), newQuantity);
+    }
```

Comment:
Your implementation of the PurchaseOperation doesn't check the balance before saving it in the Storage. According to the checklist, the balance should be positive, and you should throw a RuntimeException if the balance is negative. Instead of IllegalArgumentException on line 20-22, consider throwing a RuntimeException with an informative message.